File: ios/Plugin/CapacitorDittoPlugin.m
#import <Foundation/Foundation.h>
#import <Capacitor/Capacitor.h>

// Define the plugin using the CAP_PLUGIN Macro, and
// each method the plugin supports using the CAP_PLUGIN_METHOD macro.
CAP_PLUGIN(CapacitorDittoPlugin, "CapacitorDitto",
           CAP_PLUGIN_METHOD(initDitto, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(echo, CAPPluginReturnPromise);
)

File: ios/Plugin/CapacitorDittoPlugin.swift
import Foundation
import Capacitor
import DittoSwift

@objc(CapacitorDittoPlugin)
public class CapacitorDittoPlugin: CAPPlugin {
    private let implementation = CapacitorDitto()
    
    @objc func initDitto(_ call: CAPPluginCall) {
        
    }

    @objc func echo(_ call: CAPPluginCall) {
        let identity = call.getObject("identity")
        
        if (identity == nil) {
            call.reject("No identity given")
            return
        }
        
        let value = call.getString("value") ?? ""
        call.resolve([
            "value": implementation.echo(value)
        ])
    }
}
